apps/frontend/src/libs/hooks/use-table-pagination/use-table-pagination.ts

```diff
@@ -0,0 +1,84 @@
+import {
+	useCallback,
+	useEffect,
+	useMemo,
+	useState,
+} from "~/libs/hooks/hooks.js";
+
+type Parameters = {
+	startingPage: number;
+	startingRowsPerPage: number;
+	totalItems: number;
+};
+
+type ReturnType = {
+	changeRowsPerPage: (value: number) => void;
+	openFirstPage: () => void;
+	openLastPage: () => void;
+	openNextPage: () => void;
+	openPreviousPage: () => void;
+	page: number;
+	rowsPerPage: number;
+};
+
+const FIRST_PAGE = 1;
+const PAGE_INCREMENT = 1;
+
+const useTablePagination = ({
+	startingPage,
+	startingRowsPerPage,
+	totalItems,
+}: Parameters): ReturnType => {
+	const [page, setPage] = useState<number>(startingPage);
+	const [rowsPerPage, setRowsPerPage] = useState<number>(startingRowsPerPage);
+
+	const totalPages = useMemo(
+		() => Math.ceil(totalItems / rowsPerPage),
+		[totalItems, rowsPerPage],
+	);
+
+	useEffect(() => {
+		if (page > totalPages) {
+			setPage(totalPages);
+		}
+
+		if (page < FIRST_PAGE) {
+			setPage(FIRST_PAGE);
+		}
+	}, [totalPages]);
```

Comment:
we should prevent from setting invalid value to page, not change it after it was set
```suggestion
```

Reply:
I think if we control page in this component, this shouldn't be needed now